android-core/src/main/java/com/mparticle/internal/UploadHandler.java

```diff
@@ -145,8 +151,19 @@ public void handleMessageImpl(Message msg) {
                             }
                         }
                     }
+                    if (!anythingToUpload && (!mAppStateManager.getSession().isActive() || mAppStateManager.isBackgrounded())) {
+                        //If there isn't anything to upload and is in background or inactive, cancel schedule job until a new message is stored
+                        JobSchedulerUtilsKt.cancelScheduledUploadBatchJob(mContext);
+                    }
                     if (mAppStateManager.getSession().isActive() && uploadInterval > 0 && msg.arg1 == 0) {
-                        this.sendEmptyDelayed(UPLOAD_MESSAGES, uploadInterval);
+                        JobSchedulerUtilsKt.scheduleBatchUploading(mContext,
+                                uploadInterval,
+                                SchedulingBatchingType.PERIODIC,
+                                true,
+                                delay -> {
+                                    sendEmptyDelayed(UPLOAD_MESSAGES, uploadInterval);
```

Comment:
there's something funky here. I tested by setting the upload interval to 30 seconds and set the SDK to production mode. When using the legacy upload mode -  i'm not seeing uploads every 30s - it seems to be much quicker.